server/socket/handlers.js

```diff
@@ -20,87 +23,169 @@ const disconnect = (io, socket) => () => {
+      params: [],
+    }
+    execute(io, room, request);
+  }
+  io.emit('publicGames', gameController.getPublicGames());
+}
+
 // Propagate players chats & chat notifications to all in gameID
 const chat = io => (gameID, type, text, author) => {
   if (process.env.NODE_ENV !== "production") {
     console.log(`Received ${type} in game ${gameID} from ${author}: ${text}`);
   }
-  io.in(gameID).emit("chat", type, text, author);
+  io.to(gameID).emit("chat", type, text, author);
 };
 
 // Guesser's turn: Submit selected word
 const guesserNextMove = io => (gameID, player, word) => {
-  console.log(gameController.globalState)
-
-  if (gameController.globalState[gameID]) {
-    let gameState = {}
-    let thrownError
-
-    try {
-      gameState = gameController.globalState[
-        gameID
-      ].gameEngine.guesserNextMove(player, word)
-    } catch (err) {
-      console.log(err)
-      thrownError = err.message
+  const room = gameController.globalState[gameID];
+  if (room) {
+    const game = room.gameEngine;
+    const request = {
+      method: game.guesserNextMove,
+      params: [player, word],
+      word: word,
     }
-
-    io.to(gameID).emit('update', gameState, thrownError)
+    execute(io, room, request);
   }
 }
 
 // Spymaster's turn: Submit hint and number of guesses
 const spyNextMove = io => (gameID, player, hint, guesses) => {
-  if (gameController.globalState[gameID]) {
-    let gameState = {}
-    let thrownError
-
-    try {
-      gameState = gameController.globalState[gameID].gameEngine.spyNextMove(
-        player,
-        hint,
-        guesses
-      )
-    } catch (err) {
-      thrownError = err.message
+  const room = gameController.globalState[gameID];
+  if (room) {
+    const game = room.gameEngine;
+    const request = {
+      method: game.spyNextMove,
+      params: [player, hint, guesses],
     }
-
-    io.to(gameID).emit('update', gameState, thrownError)
+    execute(io, room, request);
   }
 }
 
 const endTurn = io => gameID => {
-  if (gameController.globalState[gameID]) {
-    io.to(gameID).emit(
-      'update',
-      gameController.globalState[gameID].gameEngine.endTurn()
-    )
+  const room = gameController.globalState[gameID];
+  if (room) {
+    const game = room.gameEngine;
+    const request = {
+      method: game.endTurn,
+      params: [],
+    }
+    execute(io, room, request);
   }
 }
 
 const restartGame = io => gameID => {
-  if (gameController.globalState[gameID]) {
-    io.to(gameID).emit(
-      'update',
-      gameController.globalState[gameID].gameEngine.restart()
-    )
+  const room = gameController.globalState[gameID];
+  if (room) {
+    const game = room.gameEngine;
+    const request = {
+      method: game.restart,
+      params: [],
+    }
+    execute(io, room, request);
   }
 }
 
+const refresh = io => () => {
+  io.emit('publicGames', gameController.getPublicGames());
+}
+
 module.exports = {
   disconnecting,
   disconnect,
   leave,
+  assign,
```

Comment:
should we refactor handlers into files by their types? i.e., a file for chat handlers, a file for gameplay handlers, etc.

Reply:
This is a good idea.

Reply:
Resolved by 481f105